htlcswitch/queue.go

```diff
@@ -17,34 +17,36 @@ import (
 // to signal the number of slots available, and a condition variable to allow
 // the packetQueue to know when new items have been added to the queue.
 type packetQueue struct {
-	// queueLen is an internal counter that reflects the size of the queue
-	// at any given instance. This value is intended to be use atomically
-	// as this value is used by internal methods to obtain the length of
-	// the queue w/o grabbing the main lock. This allows callers to avoid a
-	// deadlock situation where the main goroutine is attempting a send
-	// with the lock held.
-	queueLen int32
+	queue []*htlcPacket
```

Comment:
Should we add any comments explaining why/warning that the order of these are important for alignment?

Reply:
Eh, I'm not sure it's necessary. Typically in the codebase we group attributes of a struct logically (things that are utilized together are laid out close together in the struct). The new version of the structure _mostly_ adheres to this, just needed to swap some things to get rid of bad alignment. 

Usage of atomics within the codebase is _definitely_ something that should be high lighted in our code contribution guidelines. Or just a new document going over common pitfalls (like referencing a range variable in a closure, etc).